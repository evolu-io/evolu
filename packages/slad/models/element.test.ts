import {
  normalizeEditorElement,
  EditorElement,
  isNormalizedEditorElement,
} from './element';

test('normalizeEditorElement merges adjacent strings', () => {
  const element: EditorElement = {
    children: [
      'a',
      'b',
      {
        children: [
          'a',
          'b',
          'c',
          {
            children: ['a', {}, 'b', 'c'],
          },
        ],
      },
      'a',
      '',
      'b',
    ],
  };
  expect(normalizeEditorElement(element)).toMatchSnapshot();
});

test('isNormalizedEditorElement', () => {
  expect(
    isNormalizedEditorElement({
      children: ['a'],
    }),
  ).toBe(true);
  expect(
    isNormalizedEditorElement({
      children: [''],
    }),
  ).toBe(true);
  expect(
    isNormalizedEditorElement({
      children: ['a', 'b'],
    }),
  ).toBe(false);
  expect(
    isNormalizedEditorElement({
      children: [{ children: [''] }],
    }),
  ).toBe(true);
});

test('normalizeEditorElement do not add children', () => {
  expect(normalizeEditorElement({})).toMatchSnapshot();
});

test('normalizeEditorElement do not remove children', () => {
  expect(normalizeEditorElement({ children: [] })).toMatchSnapshot();
  expect(normalizeEditorElement({ children: [''] })).toMatchSnapshot();
  expect(normalizeEditorElement({ children: ['.'] })).toMatchSnapshot();
});
