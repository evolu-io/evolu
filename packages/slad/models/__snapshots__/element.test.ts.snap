// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizeEditorElement do not add children 1`] = `Object {}`;

exports[`normalizeEditorElement do not remove children 1`] = `
Object {
  "children": Array [],
}
`;

exports[`normalizeEditorElement do not remove children 2`] = `
Object {
  "children": Array [
    "",
  ],
}
`;

exports[`normalizeEditorElement do not remove children 3`] = `
Object {
  "children": Array [
    ".",
  ],
}
`;

exports[`normalizeEditorElement merges adjacent strings 1`] = `
Object {
  "children": Array [
    "ab",
    Object {
      "children": Array [
        "abc",
        Object {
          "children": Array [
            "a",
            Object {},
            "bc",
          ],
        },
      ],
    },
    "ab",
  ],
}
`;
